tests/abc_atlas_cache/test_cache.py

```diff
@@ -0,0 +1,573 @@
+        msg = str(warn[0].message)
+        assert manifest_list[1] in msg
+        assert manifest_list[-1] in msg
+        assert 'It is possible that some data files' in msg
+        cmd = f"S3CloudCache.load_manifest('{manifest_list[1]}')"
+        assert cmd in msg
+
+    def test_load_last_manifest(self):
+        """
+        Test that load_last_manifest works
+        """
+        manifest_list = self.create_manifests(manifest_count=5)
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+
+        # check that load_last_manifest in a new cache loads the
+        # latest manifest without emitting a warning
+        with warnings.catch_warnings():
+            warnings.simplefilter("error")
+            cache.load_last_manifest()
+        assert cache.current_manifest == manifest_list[-1]
+
+        cache.load_manifest(manifest_list[2])
+
+        del cache
+
+        # check that load_last_manifest on an old cache emits the
+        # expected warning and loads the correct manifest
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+        expected = ('A more up to date version of the '
+                    f'dataset -- {manifest_list[-1]} '
+                    '-- exists online')
+        with pytest.warns(OutdatedManifestWarning,
+                          match=expected):
+            cache.load_last_manifest()
+
+        assert cache.current_manifest == manifest_list[2]
+        cache.load_manifest(manifest_list[0])
+        del cache
+
+        # repeat the above test, making sure the correct manifest is
+        # loaded again
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+        expected = ('A more up to date version of the '
+                    f'dataset -- {manifest_list[-1]} '
+                    '-- exists online')
+        with pytest.warns(OutdatedManifestWarning,
+                          match=expected):
+            cache.load_last_manifest()
+
+        assert cache.current_manifest == manifest_list[0]
+
+    def test_corrupted_load_last_manifest(self):
+        """
+        Test that load_last_manifest works when the record of the last
+        manifest has been corrupted
+        """
+        manifest_list = self.create_manifests(manifest_count=5)
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+        cache.load_manifest(manifest_list[-2])
+        fname = cache._manifest_last_used.resolve()
+        del cache
+        with open(fname, 'w') as out_file:
+            out_file.write('babababa')
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+        expected = f'Loading latest version -- {manifest_list[-1]}'
+        with pytest.warns(UserWarning, match=expected):
+            cache.load_last_manifest()
+        assert cache.current_manifest == manifest_list[-1]
+
+    def test_download_directory(self):
+        """
+        Test that S3CloudCache.download_data() correctly files in the directory
```

Comment:
Check the grammar on this docstring. I think it is supposed to be

"Test that S3CloudCache.download_data() correctly downloads files from S3"